### ISSUE TITLE
NametagEdit - Download Here

### ISSUE THREAD

**Issue by Lorenzo0111** [sgtcaze/NametagEdit#631]
While we are waiting for the new NametagEdit release you can download a beta version of the plugin here.

- [Download by clicking here.](https://github.com/Lorenzo0111/NametagEdit/releases/download/4.4.16/NametagEdit.jar)

This includes all commits from the latest release to ccf9bf19d6a7fc1f9f914aee038c3fb4b79e5209

This isn't an official release but it is stable and the source code is on the [dev2](https://github.com/sgtcaze/NametagEdit/tree/dev2) branch of this repo.

**Comment by Lorenzo0111** [comment]
@Aurelien30000 can you pin this?

**Comment by Aurelien30000** [comment]
No sorry, don't have access, @JustOneMoreBlock please ;)

**Comment by BlackRec** [comment]
Are you sure you update it for 1.17.1 ? I still have some error. Could it be because of java 16 ?

```
[15:03:44 INFO]: [NametagEdit] Enabling NametagEdit v4.4.16*
[15:03:44 WARN]: java.lang.ClassNotFoundException: net.minecraft.server.v1_17_R1.PacketPlayOutScoreboardTeam
[15:03:44 WARN]: 	at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:146)
[15:03:44 WARN]: 	at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:103)
[15:03:44 WARN]: 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)
[15:03:44 WARN]: 	at java.base/java.lang.Class.forName0(Native Method)
[15:03:44 WARN]: 	at java.base/java.lang.Class.forName(Class.java:375)
[15:03:44 WARN]: 	at com.nametagedit.plugin.packets.PacketAccessor.<clinit>(PacketAccessor.java:59)
[15:03:44 WARN]: 	at com.nametagedit.plugin.packets.PacketWrapper.<clinit>(PacketWrapper.java:25)
[15:03:44 WARN]: 	at com.nametagedit.plugin.NametagEdit.testCompat(NametagEdit.java:90)
[15:03:44 WARN]: 	at com.nametagedit.plugin.NametagEdit.onEnable(NametagEdit.java:40)
[15:03:44 WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[15:03:44 WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370)
[15:03:44 WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500)
[15:03:44 WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:518)
[15:03:44 WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:432)
[15:03:44 WARN]: 	at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:639)
[15:03:44 WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.init(DedicatedServer.java:306)
[15:03:44 WARN]: 	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1126)
[15:03:44 WARN]: 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316)
[15:03:44 WARN]: 	at java.base/java.lang.Thread.run(Thread.java:831)
[15:03:44 WARN]: java.lang.ClassNotFoundException: net.minecraft.server.v1_17_R1.ChatComponentText
[15:03:44 WARN]: 	at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:146)
[15:03:44 WARN]: 	at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:103)
[15:03:44 WARN]: 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)
[15:03:44 WARN]: 	at java.base/java.lang.Class.forName0(Native Method)
[15:03:44 WARN]: 	at java.base/java.lang.Class.forName(Class.java:375)
[15:03:44 WARN]: 	at com.nametagedit.plugin.packets.PacketWrapper.<clinit>(PacketWrapper.java:28)
[15:03:44 WARN]: 	at com.nametagedit.plugin.NametagEdit.testCompat(NametagEdit.java:90)
[15:03:44 WARN]: 	at com.nametagedit.plugin.NametagEdit.onEnable(NametagEdit.java:40)
[15:03:44 WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[15:03:44 WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370)
[15:03:44 WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500)
[15:03:44 WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:518)
[15:03:44 WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:432)
[15:03:44 WARN]: 	at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:639)
[15:03:44 WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.init(DedicatedServer.java:306)
[15:03:44 WARN]: 	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1126)
[15:03:44 WARN]: 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316)
[15:03:44 WARN]: 	at java.base/java.lang.Thread.run(Thread.java:831)
[15:03:44 WARN]: java.lang.NullPointerException: Cannot invoke "java.lang.Class.newInstance()" because "com.nametagedit.plugin.packets.PacketAccessor.packetClass" is null
[15:03:44 WARN]: 	at com.nametagedit.plugin.packets.PacketAccessor.createPacket(PacketAccessor.java:123)
[15:03:44 WARN]: 	at com.nametagedit.plugin.packets.PacketWrapper.<init>(PacketWrapper.java:18)
[15:03:44 WARN]: 	at com.nametagedit.plugin.NametagEdit.testCompat(NametagEdit.java:90)
[15:03:44 WARN]: 	at com.nametagedit.plugin.NametagEdit.onEnable(NametagEdit.java:40)
[15:03:44 WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[15:03:44 WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370)
[15:03:44 WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500)
[15:03:44 WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:518)
[15:03:44 WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:432)
[15:03:44 WARN]: 	at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:639)
[15:03:44 WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.init(DedicatedServer.java:306)
[15:03:44 WARN]: 	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1126)
[15:03:44 WARN]: 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316)
[15:03:44 WARN]: 	at java.base/java.lang.Thread.run(Thread.java:831)
[15:03:44 INFO]: [NametagEdit] Disabling NametagEdit v4.4.16
[15:03:44 ERROR]: Error occurred while disabling NametagEdit v4.4.16 (Is it up to date?)
java.lang.NullPointerException: Cannot invoke "com.nametagedit.plugin.NametagManager.reset()" because "this.manager" is null
	at com.nametagedit.plugin.NametagEdit.onDisable(NametagEdit.java:73) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:265) ~[patched_1.17.1.jar:git-Paper-108]
	at org.bukkit.plugin.java.JavaPluginLoader.disablePlugin(JavaPluginLoader.java:405) ~[patched_1.17.1.jar:git-Paper-108]
	at org.bukkit.plugin.SimplePluginManager.disablePlugin(SimplePluginManager.java:533) ~[patched_1.17.1.jar:git-Paper-108]
	at org.bukkit.plugin.SimplePluginManager.disablePlugin(SimplePluginManager.java:525) ~[patched_1.17.1.jar:git-Paper-108]
	at com.nametagedit.plugin.NametagEdit.testCompat(NametagEdit.java:93) ~[?:?]
	at com.nametagedit.plugin.NametagEdit.onEnable(NametagEdit.java:40) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.17.1.jar:git-Paper-108]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[patched_1.17.1.jar:git-Paper-108]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[patched_1.17.1.jar:git-Paper-108]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:518) ~[patched_1.17.1.jar:git-Paper-108]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:432) ~[patched_1.17.1.jar:git-Paper-108]
	at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:639) ~[patched_1.17.1.jar:git-Paper-108]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:306) ~[patched_1.17.1.jar:git-Paper-108]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1126) ~[patched_1.17.1.jar:git-Paper-108]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[patched_1.17.1.jar:git-Paper-108]
	at java.lang.Thread.run(Thread.java:831) [?:?]
[15:03:44 ERROR]: [NametagEdit] 
------------------------------------------------------
[WARNING] NametagEdit v4.4.16 Failed to load! [WARNING]
------------------------------------------------------
This might be an issue with reflection. REPORT this:
The plugin will now self destruct.
------------------------------------------------------
```

**Comment by Aurelien30000** [comment]
Noooo @Lorenzo0111 you built the wrong version, use dev2 branch :)

Sorry @quentin3225 and ty for info!

**Comment by Lorenzo0111** [comment]
> Noooo @Lorenzo0111 you built the wrong version, use dev2 branch :)
> 
> Sorry @quentin3225 and ty for info!

Oh no 🤦‍♂️

**Comment by Lorenzo0111** [comment]
@Aurelien30000 now I can’t. Can you compile it and put it in a GitHub release(maybe on your fork) then give me the link and I’ll update the post

**Comment by Aurelien30000** [comment]
# ! DOWNLOAD THIS FILE FOR 1.17.x SUPPORT !
![NametagEdit.zip](https://github.com/sgtcaze/NametagEdit/files/6888562/NametagEdit.zip)
# ! DOWNLOAD THIS FILE FOR 1.17.x SUPPORT !

**Comment by BlackRec** [comment]
It's working great now ! Thanks for your work ! :)

**Comment by Lorenzo0111** [comment]
I've also fixed the url in my message

**Comment by JustOneMoreBlock** [comment]
This has been resolved in #633. Closes issue.

You can download the latest version here.

https://ci.nametagedit.com/job/NametagEdit/lastSuccessfulBuild/artifact/target/NametagEdit.jar

- Supports: 1.8.x to 1.17.1
- HEX Code Support - Thanks to https://github.com/Lorenzo0111 and https://github.com/Aurelien30000
- Added `api-version`. Too many issues reported.
- Updated database to use jdbc instead.
- Updated our database to version 5. Alters the `prefix` and `suffix` tables to increase the `varchar` from 64 to 256.
- Many other bug fixes.
